src/Sylius/Bundle/ApiBundle/Controller/Payment/GetPaymentConfiguration.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller\Payment;
+
+use Sylius\Bundle\ApiBundle\Provider\PaymentConfigurationProvider;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/** @experimental */
+final class GetPaymentConfiguration
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var PaymentConfigurationProvider */
+    private $paymentConfigurationProvider;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        PaymentConfigurationProvider $paymentConfigurationProvider
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->paymentConfigurationProvider = $paymentConfigurationProvider;
+    }
+
+    public function __invoke(Request $request): JsonResponse
+    {
+        /** @var OrderInterface $order */
+        $order = $this->orderRepository->find($request->get('id'));
+
+        return new JsonResponse($this->paymentConfigurationProvider->provide($order->getLastPayment()));
```

Comment:
Is good idea to get last payment? Maybe we should be able to add needed payment `id` in the path? I continue idea from my comment below with using `/new-api/shop/orders/{id}/payments/{paymentId}/configuration`. What is your opinion?

Reply:
If we want to go with the last payment, we should use the last new payment. But your proposed path seems like a better approach